src/preview/server.ts

```diff
@@ -184,7 +196,52 @@ function sendOkResponse(response: http.ServerResponse, content: Buffer, contentT
     response.end(content)
 }
 
+async function proxyHandler(req: http.IncomingMessage, res: http.ServerResponse) {
```

Comment:
We follow a minimal intrusion principle for the extra features. It means that functions like this one should be put into `liveshare.ts` and add minimal number of changed lines to other files.